qiskit_machine_learning/kernels/quantum_kernel.py

```diff
@@ -380,38 +387,34 @@ def construct_circuit(
         psi_x = self._feature_map.assign_parameters(x_dict)
         qc.append(psi_x.to_instruction(), qc.qubits)
 
-        if not is_statevector_sim:
-            if y is not None and len(y) != self._feature_map.num_parameters:
-                raise ValueError(
-                    "y and class feature map incompatible dimensions.\n"
-                    f"y has {len(y)} dimensions, but feature map has {self._feature_map.num_parameters}."
-                )
+        return qc
 
-            if y is None:
-                y = x
+    # just a synonym
+    def _construct_circuit_statevector(self, x: ParameterVector, y: ParameterVector = None):
+        """This is just a synonym for ``_construct_circuit_with_feature_map``"""
+        return self._construct_circuit_with_feature_map(x, y)
 
-            y_dict = dict(zip(self._feature_map.parameters, y))
-            psi_y_dag = self._feature_map.assign_parameters(y_dict)
-            qc.append(psi_y_dag.to_instruction().inverse(), qc.qubits)
+    def _compute_overlap_statevector(self, idx: Tuple[int, int], results: List) -> float:
+        """
+        Helper function to compute overlap for given input if a statevector simulator is used.
+        """
+        # |<0|Psi^dagger(y) x Psi(x)|0>|^2, take the amplitude
+        v_a, v_b = [results[int(i)] for i in idx]
```

Comment:
Is the typhint for idx correct? Casting a tuple onto int would not work...

Reply:
The type hint was correct! I understand the code was confusing, I rewrote it without tuples, I hope now it is easier to understand.